src/cdk/a11y/key-manager/activedescendant-key-manager.ts

```diff
@@ -40,13 +40,18 @@ export class ActiveDescendantKeyManager<T> extends ListKeyManager<Highlightable
   setActiveItem(item: T): void;
 
   setActiveItem(index: any): void {
+    const previousIndex = this.activeItemIndex;
+
     if (this.activeItem) {
       this.activeItem.setInactiveStyles();
     }
-    super.setActiveItem(index);
+    super.updateActiveItem(index);
     if (this.activeItem) {
       this.activeItem.setActiveStyles();
     }
+    if (previousIndex !== this.activeItemIndex) {
+      this.change.next(this.activeItemIndex);
+    }
```

Comment:
This part of the change is probably okay, but changing the public API is technically a breaking change. We'd also need a unit test for this change in behavior.

Reply:
This makes it inconsistent with the other key managers (e.g. `FocusKeyManager`) which still have the old event timing. Maybe we should tackle the issue at the base class level?